tinygrad/tensor.py

```diff
@@ -1259,8 +1254,10 @@ def pad2d(self, padding:Sequence[int], value:float=0.0) -> Tensor:
     print(t.pad2d((1, 1, 2, 0), value=-float("inf")).numpy())
     ```
     """
```

Comment:
Just a fyi, currently `pad2d` does a lot of the heavy lifting `_slice` used to do. 
the documentation says `(padding_left, padding_right, padding_top, padding_bottom)` but we use `pad2d` for `len(padding) == 2 or 4 or 6` throughout the codebase.

Reply:
Why does pad2d have to shrink?

Reply:
negative padding
```
  def test_pad2d(self):
    helper_test_op([(3,3,3,3)], lambda x: torch.nn.functional.pad(x, (1,2,3,4)), lambda x: x.pad2d(padding=(1,2,3,4)))
    helper_test_op([(3,3,3,3)], lambda x: torch.nn.functional.pad(x, (-1,2,-3,4)), lambda x: x.pad2d(padding=(-1,2,-3,4)))
    helper_test_op([(3,3,3,3)], lambda x: torch.nn.functional.pad(x, (1,2,3,4), value=5), lambda x: x.pad2d(padding=(1,2,3,4),value=5))
    helper_test_op([(3,3,3,3)], lambda x: torch.nn.functional.pad(x, (-1,2,-3,4), value=5), lambda x: x.pad2d(padding=(-1,2,-3,4),value=5))
```